File: LoanCalculator/LoanApplication.swift
//
//  LoanApplication.swift
//  LoanCalculator
//
//  Created by Ivan Tishchenko on 17.12.2025.
//

import Foundation

// MARK: - Loan Application Model
struct LoanApplication: Codable {
    let amount: Double
    let period: Int
    let totalRepayment: Double
}

// MARK: - API Response Model
struct LoanApplicationResponse: Codable {
    let id: Int
    let amount: Double
    let period: Int
    let totalRepayment: Double
}

File: Android/app/src/main/java/com/loancalculator/network/LoanApplication.kt
package com.loancalculator.network

import com.google.gson.annotations.SerializedName

/**
 * Loan Application request model
 */
data class LoanApplication(
    @SerializedName("amount")
    val amount: Double,
    
    @SerializedName("period")
    val period: Int,
    
    @SerializedName("totalRepayment")
    val totalRepayment: Double
)

/**
 * Loan Application response model
 */
data class LoanApplicationResponse(
    @SerializedName("id")
    val id: Int,
    
    @SerializedName("amount")
    val amount: Double,
    
    @SerializedName("period")
    val period: Int,
    
    @SerializedName("totalRepayment")
    val totalRepayment: Double
)

